example/src/main/java/io/netty/example/http2/tiles/Http2Server.java

```diff
@@ -0,0 +1,82 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.example.http2.tiles;
+
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelFuture;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelOption;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.socket.SocketChannel;
+import io.netty.channel.socket.nio.NioServerSocketChannel;
+import io.netty.handler.codec.http2.Http2OrHttpChooser.SelectedProtocol;
+import io.netty.handler.codec.http2.Http2SecurityUtil;
+import io.netty.handler.ssl.ApplicationProtocolConfig;
+import io.netty.handler.ssl.ApplicationProtocolConfig.Protocol;
+import io.netty.handler.ssl.ApplicationProtocolConfig.SelectedListenerFailureBehavior;
+import io.netty.handler.ssl.ApplicationProtocolConfig.SelectorFailureBehavior;
+import io.netty.handler.ssl.SslContext;
+import io.netty.handler.ssl.SslProvider;
+import io.netty.handler.ssl.SupportedCipherSuiteFilter;
+import io.netty.handler.ssl.util.SelfSignedCertificate;
+
+import java.security.cert.CertificateException;
+
+import javax.net.ssl.SSLException;
+
+/**
+ * Demonstrates a Http2 server using Netty to display a bunch of images and
+ * simulate latency. It is a Netty version of the <a href="https://http2.golang.org/gophertiles?latency=0">
+ * Go lang HTTP2 tiles demo</a>.
+ */
+public class Http2Server {
+
+    public static final int PORT = Integer.parseInt(System.getProperty("http2-port", "8443"));
+    static final int MAX_CONTENT_LENGTH = 1024 * 100;
+
+    private final EventLoopGroup group;
+
+    public Http2Server(EventLoopGroup eventLoopGroup) {
+        group = eventLoopGroup;
+    }
+
+    public ChannelFuture start() throws Exception {
+        final SslContext sslCtx = configureTLS();
+        ServerBootstrap b = new ServerBootstrap();
+        b.option(ChannelOption.SO_BACKLOG, 1024);
+        b.group(group).channel(NioServerSocketChannel.class).childHandler(new ChannelInitializer<SocketChannel>() {
+            @Override
+            protected void initChannel(SocketChannel ch) throws Exception {
+                ch.pipeline().addLast(sslCtx.newHandler(ch.alloc()), new Http2OrHttpHandler(MAX_CONTENT_LENGTH));
+            }
+        });
+
+        Channel ch = b.bind(PORT).sync().channel();
+        return ch.closeFuture();
+    }
+
+    private SslContext configureTLS() throws CertificateException, SSLException {
+        SelfSignedCertificate ssc = new SelfSignedCertificate();
+        final SslContext sslCtx = SslContext.newServerContext(SslProvider.JDK, ssc.certificate(), ssc.privateKey(),
```

Comment:
Yes, that's exactly what I did (there's a constructor), but it chose OpenSSL and I couldn't make it work. Since I believe it will take me some time to sort that out, I thought about leaving it as an exercise to the reader :) If you know what exactly is needed to make HTTP/2 work with OpenSSL I can give it a try and see if that works for me. Even because, I believe we should put instructions for both JDK and OpenSSL in the javadocs, if we go on that direction.